services/core-web/src/components/mine/Permit/MinePermitInfo.js

```diff
@@ -160,13 +174,17 @@ export class MinePermitInfo extends Component {
 
     this.setState({ modifiedPermits: true });
 
-    return this.props.createPermit(this.props.mineGuid, payload).then(this.closePermitModal);
+    return this.props.createPermit(this.props.mineGuid, payload).then((data) => {
+      const siteProperties = { ...values.site_properties, permit_guid: data.data.permit_guid };
+      this.props.createMineTypes(this.props.mineGuid, [siteProperties]).then(this.closePermitModal);
+    });
   };
 
-  handleEditPermit = (values) =>
-    this.props
+  handleEditPermit = (values) => {
+    return this.props
```

Comment:
Why change to explicit return? I think we prefer implicit returns wherever possible

Reply:
I likely added a console.log and forgot to remove the block after. I'll fix this